src/coreComponents/dataRepository/Group.hpp

```diff
@@ -509,11 +550,14 @@ class Group
     localIndex counter = 0;
     for( auto const & subgroup : subGroupKeys )
     {
-      applyLambdaToContainer< GROUPTYPE, GROUPTYPES... >( getGroup( subgroup ), [&]( auto & castedSubGroup )
+      if( hasGroup( subgroup ) )
```

Comment:
I'm must confess I'm a bit confused by this new check.
We use to have a `Group` (and therefore the whole `GEOSX`) failing in case a `Group` instance "_was not there_" (the call to `getGroup` would `throw`/`exit`).

Now we silently accept the sub-group not being there.
There surely is a reason, but wouldn't this open more space for vicious bugs?

Reply:
oof. I think I put this in as a temporary workaround...and was supposed to remove this. nice catch.